src/main/java/uk/gov/justice/hmpps/prison/aop/connectionproxy/OracleConnectionAspect.java

```diff
@@ -40,12 +42,15 @@ public OracleConnectionAspect(
 
     @Override
     protected Connection openProxySessionIfIdentifiedAuthentication(final Connection pooledConnection) throws SQLException {
-        final var proxyUserAuthSource = authenticationFacade.getProxyUserAuthenticationSource();
-        if (proxyUserAuthSource == NOMIS) {
+        if (!RoutingDataSource.isReplica()) {
+            clearContext(pooledConnection);
+        }
```

Comment:
We are working on the assumption that clearing the context does not cause a performance overhead - prepared to revert if this proves false.